api.js

```diff
@@ -224,3 +236,87 @@ Api.prototype._run = function (files, _options) {
 		return runStatus;
 	});
 };
+
+var getBlankResults = function () {
+	return {
+		stats: {
+			testCount: 0,
+			passCount: 0,
+			skipCount: 0,
+			todoCount: 0,
+			failCount: 0
+		},
+		tests: []
+	};
+};
+
+Api.prototype._runLimitedPool = function (files, runStatus, concurrency) {
+	var self = this;
+	var tests = {};
+
+	runStatus.on('timeout', function () {
+		Object.keys(tests).forEach(function (file) {
+			var fork = tests[file];
+			fork.exit();
+		});
+	});
+
+	return Promise.map(files, function (file) {
+		try {
+			var test = tests[file] = self._runFile(file, runStatus);
+
+			return new Promise(function (resolve) {
+				var runner = self._generateRunner(file, test, runStatus, resolve);
+				test.on('stats', runner);
+				test.on('exit', function () {
+					delete tests[file];
+				});
+				test.catch(runner);
+			});
+		} catch (err) {
+			runStatus.handleExceptions({
+				exception: err,
+				file: path.relative('.', file)
+			});
+
+			return getBlankResults();
+		}
+	}, {concurrency: concurrency})
+		.then(function (results) {
+			// cancel debounced _onTimeout() from firing
+			if (self.options.timeout) {
+				runStatus._restartTimer.cancel();
+			}
+
+			runStatus.processResults(results);
+			return runStatus;
+		});
+};
+
+Api.prototype._generateRunner = function (file, test, runStatus, cb) {
+	var self = this;
+	return function () {
+		self.emit('ready');
```

Comment:
This probably shouldn't be emitted for every fork. Doesn't look like it's used for anything though.